### PR TITLE
Refactor minimega code base to use Go Modules

### PR DESCRIPTION
**Pull request by activeshadow**
This PR replaces #1427.

### CODE REVIEW

**Review by aherna — CHANGES_REQUESTED** [review]

**Comment by activeshadow** [comment]
@aherna any movement on this? Would be great to get this merged!

**Comment by aherna** [comment]
> @aherna any movement on this? Would be great to get this merged!

Were the issues resolved? Ie all.bash throwing out errors when vetting

**Comment by activeshadow** [comment]
> > @aherna any movement on this? Would be great to get this merged!
> 
> @activeshadow Were the issues resolved? Ie all.bash throwing out errors when vetting

@aherna I just ran the `all.bash` script and everything worked fine. I'm using Go version 1.17. What version are you testing with? It may be that I need to leave both versions of build directives in the code to support older versions of Go.

**Comment by activeshadow** [comment]
@aherna okay I've fixed the issue you found (was due to older versions of Go not knowing about the new `go:build` directive). Let's get this merged in shall we?! :+1::joy:

File: cmd/miniccc/process_windows.go [hunk 1]
```diff
@@ -2,7 +2,7 @@
 // Under the terms of Contract DE-NA0003525 with NTESS, the U.S. Government retains certain
 // rights in this software.
 
-// +build windows
+//go:build windows
 
 package main
```
[the remark this answers is not in the document]
Comment (activeshadow) [reply cmd/miniccc/process_windows.go]: This has been addressed.

**Comment by activeshadow** [comment]
@aherna anything new with this?

**Comment by activeshadow** [comment]
@aherna FYSA, I've been running this branch locally and for some other customers and all is working as expected.

**Comment by activeshadow** [comment]
@aherna can we go ahead and get this merged by chance? I'd like to use it as the basis for the fixes and updates we're currently working on.

**Review by aherna — APPROVED** [review]
LGTM

**Pull request merged**